### ISSUE TITLE
Private Git repos failing

### ISSUE THREAD

**Issue by dhess** [NixOS/nix#3508]
With the latest Hydra rev in `nixpkgs-unstable` (https://github.com/NixOS/nixpkgs-channels/commit/342eaba9ebbc68923c1028a56be1c94e74862832), any job that fetches from a private repo is erroring out during Hydra evaluation, apparently because the project revs aren't being parsed correctly:

```
hydra-eval-jobs returned exit code 1:
trace: Using latest index state for compendium!
warning: dumping very large path (> 256 MiB); this may run out of memory
trace: Using index-state: 2020-04-07T00:00:00Z for compendium
trace: Using latest index state for compendium!
trace: Using index-state: 2020-04-07T00:00:00Z for compendium
error: "invalid Git branch/tag name '*'"
error: unexpected EOF reading a line
```

Based on the "invalid Git branch/tag name" error and some searching across nixpkgs, Hydra, and Nix, this appears to be a Nix issue, though it's hard to be certain.

These projects were working fine prior to the most recent . I've done the recent Hydra migration, converted the URLs from `git@github.com:...` to the new `ssh://git@github.com/...` format, etc. 

Hydra reports the following version:

`Hydra 2020-04-07 (using nix-2.4pre20200403_3473b19)`

**Comment by dhess** [comment]
I had to revert to `Hydra 2020-03-24 (using nix-2.4pre7250_94c93437)` (using `nixUnstable`) to get things working again.

**Comment by domenkozar** [comment]
See also https://github.com/NixOS/nix/issues/3503

**Comment by dhess** [comment]
@domenkozar Yeah, I've already made those changes to my projects. This is a separate issue.

**Comment by AleXoundOS** [comment]
Unfortunately. Old nix version that supports ssh does not support git submodules. New one that supports git submodules does not support ssh.

**Comment by AleXoundOS** [comment]
I guess the parsing [URL](https://github.com/NixOS/nix/blob/5596f879b43c20e5115d7bf9fddb70cdeec78988/src/libutil/url.cc#L13) or [URI](https://github.com/NixOS/nix/blob/5596f879b43c20e5115d7bf9fddb70cdeec78988/src/libutil/url.hh#L33) regexps should be modified to allow a colon after domain.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
